frontends/common/constantFolding.cpp

```diff
@@ -559,9 +557,7 @@ const IR::Node* DoConstantFolding::postorder(IR::Slice* e) {
     big_int mask = 1;
     mask = (mask << (m - l + 1)) - 1;
     value = value & mask;
-    auto resultType = typeMap->getType(getOriginal(), true);
-    if (!resultType->is<IR::Type_Bits>())
-        BUG("Type of slice is not Type_Bits, but %1%", resultType);
+    auto resultType = IR::Type_Bits::get(m - l + 1);
```

Comment:
Should this be signed or unsigned?  This uses unsigned unconditionaly, but if the constant is a signed should it be signed?  I forget what we decided to say in the spec, but the old code would preserve the signedness of the source

Reply:
This is from the 1.2.0 spec changelog:
```
Define the type of bit slices from signed and unsigned values to be unsigned. 
```